backend/src/main/java/io/debezium/configserver/rest/ConnectorResource.java

```diff
@@ -56,6 +61,9 @@ public class ConnectorResource {
 
     private static final Logger LOGGER = Logger.getLogger(ConnectorResource.class);
 
+    @Inject
+    JolokiaClient jolokiaClient;
```

Comment:
If we were talking about some other dependency injection frameworks, perhaps, but Quarkus simplifies this to the point that the choice you make here is irrelevant as you can use the same test injection style regardless if the bean uses `@Inject`, constructor, or setter -based injection:

```
@BeforeAll
public static void applyJolokiaClientMockForTests() {
    JolokiaClient client = Mockito.mock(JolokiaClient.class);
    // Mockito lines to build the client stub we want for tests
    QuarkusMock.installMockForType(client, JolokiaClient.class)
}
```

I think since we've mostly (if not completely used) the style @ani-sha used here, I'd favor consistency since it really doesn't impact the testing approach we'd need to use for Quarkus.

Reply:
Hey, I thought about this over the weekend. While this is easy to test with Quarkus, I think in general such kind of injection is often seen as a code smell and maybe we can simply move it to the constructor. This won't hurt, does it?

Reply:
@jcechace What do people learn at university nowadays? Is this good or bad?

Reply:
I agree with @jcechace more or less. 

But let me add more on this sentence `Additionally with higher amount of dependencies the constructor parameters can often become become chaotic.`

This is exactly why it's better to use constructor injection (a part from framework agnostic), it show you , with the increase of parameters, that probably you class is doing to much stuff. Using attribute injection this usually easy less evident. For example some static code analysis has a built-in rule for checking number of parameters on constructors (https://rules.sonarsource.com/java/RSPEC-107/).

Reply:
Yes. That's exactly what I had in mind.
While practically high-level languages with frameworks like Quarkus compile and rewrite your code to proxie classes and you can access and replace basically everything... or alternatively simpler by cracking up hidden members with reflection, it's not only not elegant, it is considered bad for all those reasons.
Thanks for the confirmation all. I didn't meant to be too picky on it and wanted your opinion and experiences. 

@ani-sha in that light, could you move this to the constructor? I think you can still let Quarkus inject the instance for you like in section [5.4. Simplified Constructor Injection](https://quarkus.io/guides/cdi-reference#simplified-constructor-injection).

Reply:
> I agree with @jcechace more or less.
> 
> But let me add more on this sentence `Additionally with higher amount of dependencies the constructor parameters can often become become chaotic.`
> 
> This is exactly why it's better to use constructor injection (a part from framework agnostic), it show you , with the increase of parameters, that probably you class is doing to much stuff. Using attribute injection this usually easy less evident. For example some static code analysis has a built-in rule for checking number of parameters on constructors (https://rules.sonarsource.com/java/RSPEC-107/).

While true we have to account for java's verbose naming customs. In practice sometimes you just have the 3-5 parameters which do make a complete sense yet the method signature is already bloated -- and we have that in debezium code base as well.

Reply:
@ani-sha can you fix this, please? Move the injected dependency into the constructor, please.